modules/pubxaiAnalyticsAdapter.js

```diff
@@ -13,196 +23,335 @@ const defaultHost = 'api.pbxai.com';
 const auctionPath = '/analytics/auction';
 const winningBidPath = '/analytics/bidwon';
 
-let initOptions;
-let auctionTimestamp;
-let auctionCache = [];
-let events = {
-  bids: [],
-  floorDetail: {},
-  pageDetail: {},
-  deviceDetail: {}
+/**
+ * auctionCache is a global cache object which stores all auction histories
+ * for the session. When getting a key from the auction cache, any
+ * information already known about the auction or associated data (floor
+ * data configured by prebid, browser data, user data etc) is added to
+ * the cache automatically.
+ */
+export const auctionCache = new Proxy(
+  {},
+  {
+    get: (target, name) => {
+      if (!Object.hasOwn(target, name)) {
+        target[name] = {
+          bids: [],
+          auctionDetail: {
+            refreshRank: Object.keys(target).length,
+            auctionId: name,
+          },
+          floorDetail: {},
+          pageDetail: {
+            host: getWindowLocation().host,
+            path: getWindowLocation().pathname,
+            search: getWindowLocation().search,
+          },
+          deviceDetail: {
+            platform: navigator.platform,
+            deviceType: getDeviceType(),
+            deviceOS: getOS(),
+            browser: getBrowser(),
+          },
+          userDetail: {
+            userIdTypes: Object.keys(getGlobal().getUserIds?.() || {}),
+          },
+          pmacDetail: JSON.parse(getStorage()?.getItem('pubx:pmac')) || {}, // {auction_1: {floor:0.23,maxBid:0.34,bidCount:3},auction_2:{floor:0.13,maxBid:0.14,bidCount:2}
+          initOptions: {
+            ...initOptions,
+            auctionId: name, // back-compat
+          },
+          sentAs: [],
+        };
+      }
+      return target[name];
+    },
+  }
+);
+
+/**
+ * Fetch extra ad server data for a specific ad slot (bid)
+ * @param {object} bid an output from extractBid
+ * @returns {object} key value pairs from the adserver
+ */
+const getAdServerDataForBid = (bid) => {
+  const gptSlot = getGptSlotForAdUnitCode(bid);
+  if (gptSlot) {
+    return Object.fromEntires(
+      gptSlot.getTargetingKeys().map((key) => [key, gptSlot.getTargeting(key)])
```

Comment:
filter for relevant pubx and hb_pb keys only here

Reply:
done; we are now including only the "winning bid" hb_ keys, plus anything starting with "pubx-"